application/MainWindow.cpp

```diff
@@ -1563,95 +1491,40 @@ void MainWindow::on_actionLaunchInstanceOffline_triggered()
 void MainWindow::doLaunch(bool online, BaseProfilerFactory *profiler)
 {
 	if (!m_selectedInstance)
+	{
 		return;
+	}
 
-	// Find an account to use.
-	std::shared_ptr<MojangAccountList> accounts = MMC->accounts();
-	MojangAccountPtr account = accounts->activeAccount();
-	if (accounts->count() <= 0)
+	while (true)
```

Comment:
This is FUNDAMENTALLY BROKEN compared to the original. There is no way to cancel out of this endless loop, only by actually providing a password (WHEN the auth service works, it doesn't have to). REVERT.